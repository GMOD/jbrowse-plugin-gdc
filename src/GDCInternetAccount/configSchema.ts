import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { Instance } from 'mobx-state-tree'
import PluginManager from '@jbrowse/core/PluginManager'
import { createInternetAccountConfig } from '@jbrowse/core/pluggableElementTypes/models'

function GDCConfigFactory(pluginManager: PluginManager) {
  return ConfigurationSchema(
    'GDCInternetAccount',
    {
      validDomains: {
        description:
          'array of valid domains the url can contain to use this account. Empty = all domains',
        type: 'stringArray',
        defaultValue: [],
      },
      authHeader: {
        description: 'custom auth header for authorization',
        type: 'string',
        defaultValue: 'Authorization',
      },
      customEndpoint: {
        description: 'custom endpoint for the external token resource',
        type: 'string',
        defaultValue: '',
      },
    },
    {
      baseConfiguration: createInternetAccountConfig(pluginManager),
      explicitlyTyped: true,
    },
  )
}

export type GDCInternetAccountConfigModel = ReturnType<typeof GDCConfigFactory>
export type OAuthInternetAccountConfig = Instance<GDCInternetAccountConfigModel>
export default GDCConfigFactory
