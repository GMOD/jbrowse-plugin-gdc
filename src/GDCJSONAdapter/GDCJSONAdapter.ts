import {
    BaseFeatureDataAdapter,
    BaseOptions,
  } from '@jbrowse/core/data_adapters/BaseAdapter'
  import { FileLocation, Region } from '@jbrowse/core/util/types'
  import { openLocation } from '@jbrowse/core/util/io'
  import { ObservableCreate } from '@jbrowse/core/util/rxjs'
  import SimpleFeature, { Feature } from '@jbrowse/core/util/simpleFeature'
  import { readConfObject } from '@jbrowse/core/configuration'
  import { AnyConfigurationModel } from '@jbrowse/core/configuration/configurationSchema'
  import GDCFeature from '../GDCAdapter/GDCFeature'
import { isArray } from '@material-ui/data-grid'
  
  export default class GDCJSONAdapter extends BaseFeatureDataAdapter {
    public static capabilities = ['getFeatures', 'getRefNames']
  
    private featureType: string

    private data: string
  
    public constructor(config: AnyConfigurationModel) {
      super(config)
      const featureType = readConfObject(config, 'featureType') as string
      const data = readConfObject(this.config, 'data') as string

      this.featureType = featureType
      this.data = data
    }

    private convertPropertyCaseToCamel(src: any) {
      const tgt = src
      Object.keys(src).forEach((k) => {
        const toCamel = (str:any) => {
          return str.replace(/([-_][a-z])/ig, ($1:any) => {
            return $1.toUpperCase()
              .replace('_', '');
          });
        };

        if (!isArray(src[k])) {
          tgt[toCamel(k)] = src[k]
          delete tgt[k]
        }
      })

      return tgt
    }

    private constructMutation(obj: any) {
      const consequence = obj.consequence

      let edges = []
      for (const transcript of consequence) {
        edges.push({
          node: {
            ...transcript
          }
        })
      }

      const gdcObject = this.convertPropertyCaseToCamel(obj)

      const genomicDnaChange = gdcObject.genomicDnaChange
      gdcObject.chromosome = genomicDnaChange.split(":")[0]
      gdcObject.mutationType = "Simple Somatic Mutation"
      gdcObject.startPosition = parseInt(genomicDnaChange.split(".")[1].split(">")[0].slice(0, -1))
      gdcObject.endPosition = gdcObject.startPosition
      gdcObject.consequence = {
        hits: {
          edges
        }
      }

      gdcObject.nciBuild = "GRCh38"
      gdcObject.numOfCasesInCohort = 1
      gdcObject.occuranceInCohort = 1
      gdcObject.percentage = 100
      gdcObject.referenceAllele = genomicDnaChange.split(".")[1].split(">")[0].slice(-1)
      gdcObject.score = 0

      return gdcObject
    }

    private constructGene(obj: any) {
      const gdcObject = this.convertPropertyCaseToCamel(obj)
      return gdcObject
    }

    private constructFeature(obj: any, idField: string) {
      const gdcObject = idField === 'ssmId' ? this.constructMutation(obj) : this.constructGene(obj)

      return new GDCFeature({
        gdcObject,
        id: gdcObject[idField],
        featureType: this.featureType,
      })
    }

    public async getRefNames() {
      return [
        'chr1',
        'chr10',
        'chr11',
        'chr12',
        'chr13',
        'chr14',
        'chr15',
        'chr16',
        'chr17',
        'chr18',
        'chr19',
        'chr2',
        'chr20',
        'chr21',
        'chr22',
        'chr3',
        'chr4',
        'chr5',
        'chr6',
        'chr7',
        'chr8',
        'chr9',
        'chrX',
        'chrY',
      ]
    }
  
    public getFeatures(region: Region, opts: BaseOptions = {}) {
      const { refName, start, end } = region
      return ObservableCreate<Feature>(async observer => {
        try {
          const idField = this.featureType === 'mutation' ? 'ssmId' : 'geneId'
          const parsedData = JSON.parse(this.data)
          var features = parsedData.map((value:any) => {
            return this.constructFeature(value, idField)
          })

          features.forEach((feature:GDCFeature) => {
            if (
              feature.get('refName') === refName &&
              feature.get('end') > start &&
              feature.get('start') < end
            ) {
              observer.next(feature)
            }
          })
        } catch (e) {
          observer.error(e)
        }
        observer.complete()
      }, opts.signal)
    }
  
    public freeResources(): void {}
  }
  