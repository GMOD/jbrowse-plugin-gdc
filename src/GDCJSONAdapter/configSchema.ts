import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { types } from 'mobx-state-tree'

export default ConfigurationSchema(
  'GDCJSONAdapter',
  {
    featureType: {
      type: 'stringEnum',
      model: types.enumeration('Feature Type', ['mutation', 'gene']),
      defaultValue: 'mutation',
      description: 'The type of the track to add',
    },
    data: {
      type: 'string',
      defaultValue: '',
      description: 'JSON string of the GDC data from the file'
    }
  },
  { explicitlyTyped: true },
)
