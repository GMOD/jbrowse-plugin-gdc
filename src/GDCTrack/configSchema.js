import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export default pluginManager => {
    const { BaseTrackConfig: LinearGenomeTrackConfig }  = pluginManager.getPlugin('LinearGenomeViewPlugin').exports
    return ConfigurationSchema(
    'GDCTrack',
    {
      adapter: pluginManager.pluggableConfigSchemaType('adapter'),
      renderer: pluginManager.pluggableConfigSchemaType('renderer'),
    },
    { baseConfiguration: LinearGenomeTrackConfig, explicitlyTyped: true },
  )
}
