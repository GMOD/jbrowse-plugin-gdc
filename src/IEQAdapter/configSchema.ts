import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default ConfigurationSchema(
  'IeqAdapter',
  {
    ieqLocation: {
      type: 'fileLocation',
      defaultValue: { uri: '/path/to/myfile.tsv', locationType: 'UriLocation' },
    },
  },
  { explicitlyTyped: true },
)
