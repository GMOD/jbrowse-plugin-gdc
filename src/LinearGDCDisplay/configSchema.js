import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default pluginManager => {
  const { baseLinearDisplayConfigSchema } = pluginManager.getPlugin(
    'LinearGenomeViewPlugin',
  ).exports
  return ConfigurationSchema(
    'LinearGDCDisplay',
    { renderer: pluginManager.pluggableConfigSchemaType('renderer') },
    { baseConfiguration: baseLinearDisplayConfigSchema, explicitlyTyped: true },
  )
}
