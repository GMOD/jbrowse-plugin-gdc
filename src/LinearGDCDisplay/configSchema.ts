import PluginManager from '@jbrowse/core/PluginManager'
import { ConfigurationSchema } from '@jbrowse/core/configuration'
import LinearGenomeViewPlugin from '@jbrowse/plugin-linear-genome-view'

export default (pluginManager: PluginManager) => {
  const { baseLinearDisplayConfigSchema } = (
    pluginManager.getPlugin('LinearGenomeViewPlugin') as LinearGenomeViewPlugin
  ).exports
  return ConfigurationSchema(
    'LinearGDCDisplay',
    { renderer: pluginManager.pluggableConfigSchemaType('renderer') },
    { baseConfiguration: baseLinearDisplayConfigSchema, explicitlyTyped: true },
  )
}
