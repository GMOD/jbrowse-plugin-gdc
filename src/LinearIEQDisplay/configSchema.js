import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default pluginManager => {
  const { baseLinearDisplayConfigSchema } = pluginManager.getPlugin(
    'LinearGenomeViewPlugin',
  ).exports
  return ConfigurationSchema(
    'LinearIEQDisplay',
    { renderer: pluginManager.pluggableConfigSchemaType('renderer') },
    { baseConfiguration: baseLinearDisplayConfigSchema, explicitlyTyped: true },
  )
}
