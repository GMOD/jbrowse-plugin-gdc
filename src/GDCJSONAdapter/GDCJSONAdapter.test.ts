import { toArray } from 'rxjs/operators'
import GDCJSONAdapter from './GDCJSONAdapter'
import configSchema from './configSchema'
import {
  mutationsJSON,
  genesJSON,
  gdcResponse,
} from './test_data/json_test_data.js'
import fetchMock from 'jest-fetch-mock'
import { firstValueFrom } from 'rxjs'

fetchMock.enableMocks()

test('adapter can fetch features from a mutations explore json file', async () => {
  const adapter = new GDCJSONAdapter(
    configSchema.create({
      featureType: 'mutation',
      data: JSON.stringify(mutationsJSON),
    }),
  )

  const features = adapter.getFeatures({
    assemblyName: 'volvox',
    refName: 'chr3',
    start: 0,
    end: 124925000,
  })

  const names = await adapter.getRefNames()
  expect(names).toMatchSnapshot()

  const featuresArray = await firstValueFrom(features.pipe(toArray()))
  expect(featuresArray.slice(0, 3)).toMatchSnapshot()
})

test('adapter can fetch featues from a genes explore json file', async () => {
  const adapter = new GDCJSONAdapter(
    configSchema.create({
      featureType: 'gene',
      data: JSON.stringify(genesJSON),
    }),
  )

  const features = adapter.getFeatures({
    assemblyName: 'volvox',
    refName: 'chr5',
    start: 142770384,
    end: 143229011,
  })

  fetchMock.mockResponseOnce(JSON.stringify(gdcResponse))

  const featuresArray = await firstValueFrom(features.pipe(toArray()))
  expect(featuresArray.slice(0, 3)).toMatchSnapshot()
})
