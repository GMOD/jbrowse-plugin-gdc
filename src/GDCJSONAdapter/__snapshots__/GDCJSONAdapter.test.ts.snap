// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adapter can fetch featues from a genes explore json file 1`] = `Array []`;

exports[`adapter can fetch features from a mutations explore json file 1`] = `
Array [
  "chr1",
  "chr10",
  "chr11",
  "chr12",
  "chr13",
  "chr14",
  "chr15",
  "chr16",
  "chr17",
  "chr18",
  "chr19",
  "chr2",
  "chr20",
  "chr21",
  "chr22",
  "chr3",
  "chr4",
  "chr5",
  "chr6",
  "chr7",
  "chr8",
  "chr9",
  "chrX",
  "chrY",
]
`;

exports[`adapter can fetch features from a mutations explore json file 2`] = `
Array [
  Object {
    "chromosome": "chr3",
    "consequence": Object {
      "hits": Object {
        "edges": Array [
          Object {
            "node": Object {
              "transcript": Object {
                "annotation": Object {
                  "polyphen_impact": "",
                  "sift_impact": "",
                  "vep_impact": "MODIFIER",
                },
                "consequence_type": "non_coding_transcript_exon_variant",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "possibly_damaging",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": true,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "benign",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "annotation": Object {
                  "polyphen_impact": "",
                  "sift_impact": "",
                  "vep_impact": "MODIFIER",
                },
                "consequence_type": "non_coding_transcript_exon_variant",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
        ],
      },
    },
    "end": 69739599,
    "endPosition": 69739599,
    "genomicDnaChange": "chr3:g.69739599T>C",
    "mutationSubtype": "Single base substitution",
    "mutationType": "Simple Somatic Mutation",
    "nciBuild": "GRCh38",
    "numOfCasesInCohort": 31,
    "occurrenceInCohort": "1 / 31",
    "percentage": 100,
    "refName": "chr3",
    "referenceAllele": "T",
    "score": 1,
    "ssmId": "f7c735a1-1aad-529e-86ba-b694cc0243c3",
    "start": 69739598,
    "startPosition": 69739599,
    "type": "Simple Somatic Mutation",
    "uniqueId": "f7c735a1-1aad-529e-86ba-b694cc0243c3",
  },
  Object {
    "chromosome": "chr3",
    "consequence": Object {
      "hits": Object {
        "edges": Array [
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "probably_damaging",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "possibly_damaging",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "probably_damaging",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "possibly_damaging",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "possibly_damaging",
                  "sift_impact": "deleterious",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "unknown",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "possibly_damaging",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "possibly_damaging",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "probably_damaging",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "possibly_damaging",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": true,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "benign",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000114861",
                  "symbol": "FOXP1",
                },
                "is_canonical": false,
              },
            },
          },
        ],
      },
    },
    "end": 71198379,
    "endPosition": 71198379,
    "genomicDnaChange": "chr3:g.71198379C>T",
    "mutationSubtype": "Single base substitution",
    "mutationType": "Simple Somatic Mutation",
    "nciBuild": "GRCh38",
    "numOfCasesInCohort": 31,
    "occurrenceInCohort": "1 / 31",
    "percentage": 100,
    "refName": "chr3",
    "referenceAllele": "C",
    "score": 1,
    "ssmId": "73f3f305-5aac-593c-93cb-fe02378d0d3a",
    "start": 71198378,
    "startPosition": 71198379,
    "type": "Simple Somatic Mutation",
    "uniqueId": "73f3f305-5aac-593c-93cb-fe02378d0d3a",
  },
  Object {
    "chromosome": "chr3",
    "consequence": Object {
      "hits": Object {
        "edges": Array [
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M52L",
                "annotation": Object {
                  "polyphen_impact": "benign",
                  "sift_impact": "deleterious",
                  "vep_impact": "MODERATE",
                },
                "consequence_type": "missense_variant",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M37L",
                "annotation": Object {
                  "polyphen_impact": "possibly_damaging",
                  "sift_impact": "deleterious",
                  "vep_impact": "MODERATE",
                },
                "consequence_type": "missense_variant",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M52L",
                "annotation": Object {
                  "polyphen_impact": "benign",
                  "sift_impact": "deleterious_low_confidence",
                  "vep_impact": "MODERATE",
                },
                "consequence_type": "missense_variant",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M37L",
                "annotation": Object {
                  "polyphen_impact": "benign",
                  "sift_impact": "deleterious",
                  "vep_impact": "MODERATE",
                },
                "consequence_type": "missense_variant",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "unknown",
                  "sift_impact": "deleterious",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "",
                  "sift_impact": "",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M53L",
                "annotation": Object {
                  "polyphen_impact": "benign",
                  "sift_impact": "deleterious",
                  "vep_impact": "MODERATE",
                },
                "consequence_type": "missense_variant",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": true,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "annotation": Object {
                  "polyphen_impact": "",
                  "sift_impact": "",
                  "vep_impact": "MODIFIER",
                },
                "consequence_type": "non_coding_transcript_exon_variant",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M1?",
                "annotation": Object {
                  "polyphen_impact": "unknown",
                  "sift_impact": "deleterious",
                  "vep_impact": "HIGH",
                },
                "consequence_type": "start_lost",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
          Object {
            "node": Object {
              "transcript": Object {
                "aa_change": "M53L",
                "annotation": Object {
                  "polyphen_impact": "benign",
                  "sift_impact": "deleterious",
                  "vep_impact": "MODERATE",
                },
                "consequence_type": "missense_variant",
                "gene": Object {
                  "gene_id": "ENSG00000187098",
                  "symbol": "MITF",
                },
                "is_canonical": false,
              },
            },
          },
        ],
      },
    },
    "end": 69879186,
    "endPosition": 69879186,
    "genomicDnaChange": "chr3:g.69879186A>C",
    "mutationSubtype": "Single base substitution",
    "mutationType": "Simple Somatic Mutation",
    "nciBuild": "GRCh38",
    "numOfCasesInCohort": 31,
    "occurrenceInCohort": "1 / 31",
    "percentage": 100,
    "refName": "chr3",
    "referenceAllele": "A",
    "score": 1,
    "ssmId": "0a1ffe21-3a5d-5d60-87c2-160569579b22",
    "start": 69879185,
    "startPosition": 69879186,
    "type": "Simple Somatic Mutation",
    "uniqueId": "0a1ffe21-3a5d-5d60-87c2-160569579b22",
  },
]
`;
